reference/tools/intel.rst

```diff
@@ -1,6 +1,148 @@
+
+.. note::
+
+    You can launch the ``conanintelsetvars.sh|bat`` before calling your intel compiler to build a project.
+    Conan will also call it in the conanfile ``build(self)`` method when running any command with ``self.run``.
+
+
+At first, ensure you are using a *profile* like this one:
+
+.. code-block:: text
+    :caption: *intelprofile*
+
+    [settings]
+    ...
+    compiler=intel-cc
+    compiler.mode=dpcpp
+    compiler.version=2021.3
+    compiler.libcxx=libstdc++
+    build_type=Release
+
+    [buildenv]
+    CC=dpcpp
+    CXX=dpcpp
+
+    [conf]
+    tools.intel:installation_path=/opt/intel/oneapi
+
+
+The ``IntelCC`` generator can be used by name in conanfiles:
+
+.. code-block:: python
+    :caption: *conanfile.py*
+
+    class Pkg(ConanFile):
+        generators = "IntelCC"
+
+
+.. code-block:: text
+    :caption: *conanfile.txt*
+
+    [generators]
+    IntelCC
+
+And it can also be fully instantiated in the conanfile ``generate()`` method:
+
+.. code-block:: python
+    :caption: *conanfile.py*
+
+    from conan import ConanFile
+    from conan.tools.intel import IntelCC
+
+    class App(ConanFile):
+        settings = "os", "arch", "compiler", "build_type"
+
+        def generate(self):
+            intelcc = IntelCC(self)
+            intelcc.generate()
+
+
+Now, running the command :command:`conan install . -pr intelprofile` generates the ``conanintelsetvars.sh|bat`` script
+which runs the Intel *setvars* script and loads all the variables into your local environment.
+
+
+Custom configurations
+++++++++++++++++++++++
+
+Apply different installation paths and command arguments simply by changing the ``[conf]`` entries. For instance:
+
+.. code-block:: text
+    :caption: intelprofile
+
+    [settings]
+    ...
+    compiler=intel-cc
+    compiler.mode=dpcpp
+    compiler.version=2021.3
+    compiler.libcxx=libstdc++
+    build_type=Release
+
+    [buildenv]
+    CC=dpcpp
+    CXX=dpcpp
+
+    [conf]
+    tools.intel:installation_path=/opt/intel/oneapi
+    tools.intel:setvars_args=--config="full/path/to/your/config.txt" --force
+
+Run again a :command:`conan install . -pr intelprofile`, then the ``conanintelsetvars.sh`` script (if we are using Linux OS)
+will contain something like:
+
+.. code-block:: bash
+    :caption: conanintelsetvars.sh
+
+    . "/opt/intel/oneapi/setvars.sh" --config="full/path/to/your/config.txt" --force
+
+
+Reference
+---------
+
+.. currentmodule:: conan.tools.intel
+
+.. autoclass:: IntelCC
+    :members:
+
+
+conf
+++++
+
+``IntelCC`` uses these :ref:`configuration entries <reference_config_files_global_conf>`:
 
-TBD
+- ``tools.intel:installation_path``: **(required)** argument to tells Conan the installation path, if it's not defined,
+  Conan will try to find it out automatically.
+- ``tools.intel:setvars_args``: **(optional)** it is used to pass whatever we want as arguments to our `setvars.sh|bat` file.
```

Comment:
it's a bit unclear. can you put some link here? `whatever we want` doesn't say much.

Reply:
I had an URL to go there, but Intel changes the domain too frequently, so I decided to remove it.